SU2_CFD/include/numerics/turbulent/turb_sources.hpp

```diff
@@ -732,12 +733,16 @@ class CSourcePieceWise_TurbSST final : public CNumerics {
         StrainMag = PerturbedStrainMag(ScalarVar_i[0]);
       }
 
-      su2double pk = Eddy_Viscosity_i * pow(StrainMag, 2) - 2.0 / 3.0 * Density_i * ScalarVar_i[0] * diverg;
-      pk = max(0.0, min(pk, 20.0 * beta_star * Density_i * ScalarVar_i[1] * ScalarVar_i[0]));
-
       const su2double VorticityMag = GeometryToolbox::Norm(3, Vorticity_i);
       const su2double zeta = max(ScalarVar_i[1], VorticityMag * F2_i / a1);
-      su2double pw = alfa_blended * Density_i * max(pow(StrainMag, 2) - 2.0 / 3.0 * zeta * diverg, 0.0);
+
+      if (vorticity_source) {su2double p_base = Eddy_Viscosity_i * pow(VorticityMag,2); }
+      else                  {su2double p_base = Eddy_Viscosity_i * pow(StrainMag,2); }
+
+      su2double pk = p_base - 2.0 / 3.0 * Density_i * ScalarVar_i[0] * diverg;
+      pk = max(0.0, min(pk, 20.0 * beta_star * Density_i * ScalarVar_i[1] * ScalarVar_i[0]));
+
+      su2double pw = (alfa_blended * Density_i) * pk;
```

Comment:
This is not consistent with SST-1994, there is no production limiter for `w`, in SST-2003 both are limited.

Reply:
Good point.  I'm still catching up to speed on the current implementation and work.